display_list/display_list_color_filter.h

```diff
@@ -240,6 +247,65 @@ class DlLinearToSrgbGammaColorFilter final : public DlColorFilter {
   friend class DlColorFilter;
 };
 
+class DlComposedColorFilter final : public DlColorFilter {
+ public:
+  static const std::shared_ptr<DlColorFilter> makeComposed(
```

Comment:
Perhaps also an overload with std::shared_ptr types that redirects to the raw pointer version.

Reply:
Actually, reverse that. The makeComposed will eventually need to call `->shared()` on any value it gets - either to return its value, or to construct a ComposedFilter out of it.

So, have this method take shared_ptr directly and you can remove a bunch of calls to `->shared()` from the implementation.

One way or another, shared() will be called on each argument, so extracting the pointer and then remaking the shared_ptr will be a waste.

An overload for the raw pointers could be created, but then it would have to protect against null pointers, so it wouldn't be much simpler. Should we have that variant? I could go either way on that variant.